src/internal/minialloc.rs

```diff
@@ -93,11 +96,13 @@ impl<F> MiniAllocator<F> {
         self.directory.dir_entry(stream_id)
     }
 
-    fn validate(&self) -> io::Result<()> {
+    fn validate(&self, validation: Validation) -> io::Result<()> {
         let root_entry = self.directory.root_dir_entry();
         let root_stream_mini_sectors =
             root_entry.stream_len / (consts::MINI_SECTOR_LEN as u64);
-        if root_stream_mini_sectors < (self.minifat.len() as u64) {
+        if validation.is_strict()
+            && root_stream_mini_sectors < (self.minifat.len() as u64)
```

Comment:
I haven't yet had a chance to looks at this closely or inspect the example file in issue #34, but I wonder if the the extra MiniFAT entries beyond the root stream length are just filled with garbage data?  Maybe if `root_stream_mini_sectors < (self.minifat.len() as u64)` then non-strict validation should truncate `self.minifat` to a length of `root_stream_mini_sectors` entries.

Reply:
Thank you)